ntropy_sdk/ntropy_sdk.py

```diff
@@ -1609,7 +1685,120 @@ def get_account_income_report(self, account_holder_id: str) -> IncomeReport:
+
+        Returns
+        -------
+        RecurringPaymentsGroups:
+            A list of Subscription objects for this account holder's history
+        """
+
+        if not isinstance(account_holder_id, str):
+            raise ValueError("account_holder_id should be of type string")
+
+        url = f"/v2/account-holder/{account_holder_id}/recurring-payments"
+
+        recurring_payments_response = self.retry_ratelimited_request("POST", url, {})
+        data = recurring_payments_response.json()
+
+        if fetch_transactions:
+            transactions = self.get_account_holder_transactions(account_holder_id)
+            transactions_dict = {tx.transaction_id: tx for tx in transactions}
+            data = [
+                {
+                    **recurring_payments_group,
+                    "transactions": EnrichedTransactionList(
+                        [
+                            transactions_dict.get(tx_id, [])
+                            for tx_id in recurring_payments_group["transaction_ids"]
+                        ]
+                    ),
+                }
+                for recurring_payments_group in data
+            ]
+        recurring_payments_groups = RecurringPaymentsGroups(
+            sorted(
+                [RecurringPaymentsGroup.from_dict(d) for d in data],
+                key=lambda x: x.latest_payment_date,
+                reverse=True,
+            )
+        )
+
+        return RecurringPaymentsGroups(recurring_payments_groups)
+
+    def _get_account_holder_transactions(
+        self, account_holder_id: str, page=0, per_page=1000
+    ):
+        url = f"/v2/account-holder/{account_holder_id}/transactions?page={page}&per_page={per_page}"
+        try:
+            response = self.retry_ratelimited_request("GET", url, None).json()
+        except requests.HTTPError as e:
+            if e.response.status_code == 404:
+                error = e.response.json()
+                raise ValueError(f"{error['detail']}")
+            raise
+
+        return response
+
+    def get_account_holder_transactions(
+        self, account_holder_id: str
+    ) -> EnrichedTransactionList:
+        """Returns EnrichTransaction list for the account holder with the provided id
+
+        Parameters
+        ----------
+        account_holder_id : str
+            A unique identifier for the account holder.
+
+        Returns
+        -------
+        EnrichedTransactionList
+            The EnrichedTransactionList corresponding to the account holder id.
+        """
+        txs = []
+        page = 0
+        while True:
+            response = self._get_account_holder_transactions(account_holder_id, page)
+            txs += response["transactions"]
+            if response["pages"] == page + 1:
+                break
+            page += 1
+
+        return EnrichedTransactionList.from_list(
+            self, txs, [Transaction(**tx) for tx in txs]
+        )
```

Comment:
```suggestion
        page = 0
        total_pages = 1
        while page < total_pages:
            response = self._get_account_holder_transactions(account_holder_id, page)
            
            if "pages" in response and total_pages < response["pages"]:
                total_pages = response["pages"]
            
            txs += response["transactions"]
            page += 1

        return EnrichedTransactionList.from_list(
            self, txs, [Transaction(**tx) for tx in txs]
        )
```

Just a bit safer to avoid infinite loops in case we ever change the response structure

Reply:
Good catch!